packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2729,18 +2715,23 @@ class RenderFittedBox extends RenderProxyBox {
   void performLayout() {
     if (child != null) {
       child!.layout(const BoxConstraints(), parentUsesSize: true);
-      switch (fit) {
-        case BoxFit.scaleDown:
-          final BoxConstraints sizeConstraints = constraints.loosen();
-          final Size unconstrainedSize = sizeConstraints.constrainSizeAndAttemptToPreserveAspectRatio(child!.size);
-          size = constraints.constrain(unconstrainedSize);
-        case BoxFit.contain:
-        case BoxFit.cover:
-        case BoxFit.fill:
-        case BoxFit.fitHeight:
-        case BoxFit.fitWidth:
-        case BoxFit.none:
-          size = constraints.constrainSizeAndAttemptToPreserveAspectRatio(child!.size);
+      if (child!.size.isEmpty) {
+        size = constraints.smallest;
```

Comment:
The assertion was in `BoxConstraints.constrainSizeAndAttemptToPreserveAspectRatio`. I think we should try to fix that method so it can handle an empty size as input?

Reply:
I am not sure how it should behave when only one side of the target size is 0.  For example if `height` is finite while `width` is 0, Do you think `BoxConstraints.constrainSizeAndAttemptToPreserveAspectRatio` should trying to match the height(only constrain size since the aspect ratio is 0 and cannot be Preserved), or return `constraints.smallest` regardless(treat any zero size as empty)?

Reply:
Yeah I think in your example the method should return `constrainSize(inputSize)`, instead of returning `smallest` and ignoring the input size.

The reason for that is I think returning `smallest` when the child size is empty feels like a special case, one that feels a bit artificial (if we do that then the documentation needs to be updated). 
Say the input `BoxConstraints` is loose and unbounded (i.e., `BoxConstraints()`), then `RenderFittedBox` always takes the same size as its child when the child size is not empty, so ideally we don't have the special case it when the child size is empty, and always return the child size regardless if it's empty.